app/addons/documents/changes/changes-header.react.jsx

```diff
@@ -0,0 +1,236 @@
+  var ReactCSSTransitionGroup = React.addons.CSSTransitionGroup;
+
+
+  var ChangesHeader = React.createClass({
+    getInitialState: function () {
+      return {
+        showTabContent: changesFilterStore.isTabVisible()
+      };
+    },
+
+    componentDidMount: function () {
+      changesFilterStore.on('change', this.onChange, this);
+    },
+
+    onChange: function () {
+      this.setState({
+        showTabContent: changesFilterStore.isTabVisible()
+      });
+    },
+
+    toggleFilterSection: function () {
+      Actions.toggleTabVisibility();
+    },
+
+    render: function () {
+      var tabContent = '';
+
+      if (this.state.showTabContent) {
+        tabContent = <ChangesHeaderTabContent key="changesFilterSection" />;
+      }
+
+      return (
+        <div className="changes-header-section">
+          <ChangesHeaderTab onToggle={this.toggleFilterSection} />
+          <ReactCSSTransitionGroup transitionName="toggleChangesFilter" component="div" className="changes-tab-content">
+            {tabContent}
+          </ReactCSSTransitionGroup>
+        </div>
+      );
+    }
+  });
+
+  var ChangesHeaderTab = React.createClass({
+    propTypes: {
+      onToggle: React.PropTypes.func.isRequired
+    },
+
+    render: function () {
+      return (
+        <div className="dashboard-upper-menu">
+          <ul className="nav nav-tabs" id="db-views-tabs-nav">
+            <li>
+              <a href="#filter" onClick={this.props.onToggle} data-bypass="true" data-toggle="tab">
+                <i className="fonticon fonticon-plus"></i> Filter
+              </a>
+            </li>
+          </ul>
+        </div>
+      );
+    }
+  });
+
+  var ChangesHeaderTabContent = React.createClass({
+    render: function () {
+      return (
+        <div className="tab-content">
+          <div className="tab-pane active" ref="filterTab">
+            <div className="changes-header js-filter">
+              <ChangesFilter />
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
+  var ChangesFilter = React.createClass({
+    getStoreState: function () {
+      return {
+        filters: changesFilterStore.getFilters()
+      };
+    },
+
+    onChange: function () {
+      this.setState(this.getStoreState());
+    },
+
+    // props belong to the parent, so these aren't stored in the related store
+    getDefaultProps: function () {
+      return {
+        addItemPlaceholder: 'Type a filter',
+        helpText: 'e.g. debug or <1.4.1> or any regex',
+        tooltip: ''
+      };
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    submitForm: function (e) {
+      e.preventDefault();
+      e.stopPropagation();
+
+
+      var filter = this.refs.addItem.getDOMNode().value.trim();
+      if (!filter) {
+        return;
+      }
+
+      Actions.addFilterViewItem(filter);
+      this.refs.addItem.getDOMNode().value = '';
```

Comment:
Thanks, guys. I gave this one some thought and I'm kind of on the fence. I'm more than happy making to the change, but it seems to add a lot of complexity just to make it more idiomatic.

If I understand it properly, I'd need to do the following:
- new onChange handler to DOM element
- handler which passing work to actions.js
- new event added in actiontypes.js
- state of input field value stored in the store
- a handler in the store to update the value as 

vs. what is it right now:
- one line within the react view setting the input value to ''.

I'm all for keeping the code as React-y as possible, but for me, this crosses the line. Ultimately, whole need to add in an onchange handler feels like a workaround to ensure state gets modified and that we can reset the value.

[Perfectly happy being overruled, here. Just my 2 cents :)]